File: tests/test_data/multi_language/Calculator.cs
// C# test file
using System;

namespace Math
{
    public class Calculator
    {
        private double result;
        
        public Calculator()
        {
            this.result = 0.0;
        }
        
        public static double CalculateSum(double a, double b)
        {
            return a + b;
        }
        
        public Calculator Add(double value)
        {
            this.result += value;
            return this;
        }
        
        public Calculator Multiply(double value)
        {
            this.result *= value;
            return this;
        }
        
        public double GetResult()
        {
            return this.result;
        }
        
        public double Result 
        { 
            get => result; 
            set => result = value; 
        }
        
        public event Action<double> ResultChanged;
        
        protected virtual void OnResultChanged(double newResult)
        {
            ResultChanged?.Invoke(newResult);
        }
    }

    public interface IMathOperations
    {
        double Calculate(double a, double b);
    }

    public enum Operation
    {
        Add,
        Subtract,
        Multiply,
        Divide
    }

    public static class OperationExtensions
    {
        public static double Apply(this Operation op, double a, double b)
        {
            return op switch
            {
                Operation.Add => a + b,
                Operation.Subtract => a - b,
                Operation.Multiply => a * b,
                Operation.Divide => a / b,
                _ => throw new ArgumentException()
            };
        }
    }

    public struct Point
    {
        public double X { get; set; }
        public double Y { get; set; }
        
        public Point(double x, double y)
        {
            X = x;
            Y = y;
        }
        
        public double Distance() => Math.Sqrt(X * X + Y * Y);
    }
}

File: tests/test_data/multi_language/Calculator.cpp
// C++ test file
#include <iostream>
#include <memory>

namespace Math {
    template<typename T>
    class Calculator {
    private:
        T result;
    
    public:
        Calculator() : result(T{}) {}
        
        static T calculateSum(T a, T b) {
            return a + b;
        }
        
        Calculator& add(T value) {
            result += value;
            return *this;
        }
        
        Calculator& multiply(T value) {
            result *= value;
            return *this;
        }
        
        T getResult() const {
            return result;
        }
    };

    enum class Operation {
        ADD,
        SUBTRACT, 
        MULTIPLY,
        DIVIDE
    };

    template<typename T>
    T applyOperation(Operation op, T a, T b) {
        switch (op) {
            case Operation::ADD: return a + b;
            case Operation::SUBTRACT: return a - b;
            case Operation::MULTIPLY: return a * b;
            case Operation::DIVIDE: return a / b;
            default: return T{};
        }
    }
}

struct Point {
    double x, y;
    
    Point(double x = 0, double y = 0) : x(x), y(y) {}
    
    double distance() const {
        return std::sqrt(x*x + y*y);
    }
};

int main() {
    Math::Calculator<double> calc;
    auto result = calc.add(10.0).multiply(2.0).getResult();
    std::cout << "Result: " << result << std::endl;
    return 0;
}